x-pack/plugins/cases/public/components/user_actions/use_user_permissions.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useMemo, useCallback } from 'react';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import { CaseStatuses } from '../../../common/types/domain';
+
+export const useUserPermissions = ({ status }: { status?: CaseStatuses }) => {
+  const { permissions } = useCasesContext();
+  const canChangeStatus = useMemo(() => {
+    // User has full permissions
+    if (permissions.update && permissions.reopenCases) {
+      return false;
+    } else {
```

Comment:
nit: What do you think of escaping early the if statements?

Reply:
Rethought this file a bit. Hopefully it makes more sense now